leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisRegistrationStore.java

```diff
@@ -768,4 +754,273 @@ public void run() {
+        }
+
+        /**
+         * Set the key prefix for endpoint lookup by socket address.
+         * <p>
+         * Default value is {@literal EP#ADDR#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal EP:ADDR:}.
+         */
+        public Builder setEndpointBySocketAddressPrefix(String endpointBySocketAddressPrefix) {
+            this.endpointBySocketAddressPrefix = endpointBySocketAddressPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for endpoint lookup by registration identity.
+         * <p>
+         * Default value is {@literal EP#IDENTITY#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal EP:IDENTITY:}.
+         */
+        public Builder setEndpointByIdentityPrefix(String endpointByIdentityPrefix) {
+            this.endpointByIdentityPrefix = endpointByIdentityPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for endpoint locks lookup.
+         * <p>
+         * Default value is {@literal LOCK#EP#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal LOCK:EP:}.
+         */
+        public Builder setEndpointLockPrefix(String endpointLockPrefix) {
+            this.endpointLockPrefix = endpointLockPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for observation token lookup.
+         * <p>
+         * Default value is {@literal OBS#TKN#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal OBS:TKN:}.
+         */
+        public Builder setObservationTokenPrefix(String observationTokenPrefix) {
+            this.observationTokenPrefix = observationTokenPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for observation tokens list lookup by registration ID.
+         * <p>
+         * Default value is {@literal TKNS#REGID#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal TKNS:REGID:}.
+         */
+        public Builder setObservationTokensByRegistrationIdPrefix(String observationTokensByRegistrationIdPrefix) {
+            this.observationTokensByRegistrationIdPrefix = observationTokensByRegistrationIdPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key for expiration key lookup. It is a sorted set used for registration expiration (expiration date,
+         * endpoint).
+         * <p>
+         * Default value is {@literal EXP#EP}. Should not be {@code null} or empty. Leshan v1.x used {@literal EXP:EP}.
+         */
+        public Builder setEndpointExpirationKey(String endpointExpirationKey) {
+            this.endpointExpirationKey = endpointExpirationKey;
+            return this;
+        }
+
+        /**
+         * Set the clean period in seconds.
+         * <p>
+         * Default value is {@literal 60 seconds}.
+         */
+        public Builder setCleanPeriod(long cleanPeriod) {
```

Comment:
Maybe something like :

Set time between 2 periodic task about cleaning expired registration.

Reply:
I am not very familiar with the Leshan internals yet, so it's difficult for me to write descriptive JavaDocs. It was mostly a copy-paste of already existing comments.
Thank you for suggestions, helped me to better understand it.